src/ServicePulse.Host/Owin/Microsoft/StaticFileContext.cs

```diff
@@ -154,6 +158,15 @@ private void ComputeIfModifiedSince()
             }
         }
 
+        public void ApplyCacheControlHeaders()
+        {
+            if (_isStatic)
+            {
+                // 604800 = 1 week
+                _response.Headers.Set("Cache-Control", "public,max-age=604800,immutable");
```

Comment:
The intent for `immutable` is for versioned resources that really will never change. This is an improvement but the real solution would be versioned url's. Maybe that for another future release.